svi/auto_thermal_reformer/run_fullspace_sweep.py

```diff
@@ -70,47 +79,65 @@ def main(X,P):
     else:
         # If the solver didn't converge, we don't care about the solve time,
         # the objective, or any of the degree of freedom values.
-        for key in config.PARAM_SWEEP_KEYS:
-            if key not in ("X", "P", "Termination"):
+        for key in df_keys:
+            if key not in ("X", "P", "Termination", "Condition Number"):
                 df[key].append(INVALID)
 
 
 if __name__ == "__main__":
+
     argparser = config.get_sweep_argparser()
+
     argparser.add_argument(
         "--fname",
         default="fullspace-sweep.csv",
         help="Base file name for parameter sweep results"
     )
+
+    argparser.add_argument(
+        "--calc-condition-number",
+        action="store_true",
+        help="Whether to calculate the condition number or not"
+    )    
+
     args = argparser.parse_args()
+
+    df = {key: [] for key in config.PARAM_SWEEP_KEYS}
+    index_to_insert = 3
+    new_key = "Condition Number" if args.calc_condition_number else None
+    df_keys = list(df.keys())
+    if new_key:
+        df_keys.insert(index_to_insert, new_key)
+    df = {key: [] for key in df_keys}
+
     xp_samples = config.get_parameter_samples(args)
 
     fpath = os.path.join(args.data_dir, args.fname)
 
     for i, (X, P) in enumerate(xp_samples):
         print(f"Running sample {i} with X={X}, P={P}")
         try:
-            main(X,P)
+            main(X,P,args.calc_condition_number)
         except AssertionError:
             df[list(df.keys())[0]].append(X)
             df[list(df.keys())[1]].append(P)
             df[list(df.keys())[2]].append("AssertionError")
-            for key in config.PARAM_SWEEP_KEYS:
-                if key not in ("X", "P", "Termination"):
+            for key in df_keys:
+                if key not in ("X", "P", "Termination", "Condition Number"):
```

Comment:
In these cases where we hit an error, we probably do want to use `INVALID` for the condition number, right? The solution hasn't been loaded into the model, so any condition number we calculate won't be meaningful (it will just be computed at the initial guess). (Also, we're not calculating and appending condition number here, so I think this will raise an error when we try to convert to data frame.)

Reply:
Thank you for this Robert, I made the changes.